pr2eus_moveit/euslisp/robot-moveit.l

```diff
@@ -685,6 +688,21 @@
                          (/ (send acc twist-key vec-key) (expt time-scale 2))))))))
          (send pt :time_from_start (ros::time (* time-scale (send (send pt :time_from_start) :to-sec))))))
        traj))
+  (:check-state-validity ()
+    (let ((res
+           (send self :planning-environment
+                 :check-state-validity
+                 (+ moveit_msgs::PlanningSceneComponents::*ROBOT_STATE*
+                    moveit_msgs::PlanningSceneComponents::*ROBOT_STATE_ATTACHED_OBJECTS*))))
+      (when (not (send res :valid))
+        (warn ";; collision is found in the current state.~%"))
+      ;; contacts are not returned in indigo and earlier versions
+      ;; https://github.com/ros-planning/moveit/pull/2355
+      (when (send res :contacts)
+        (warn ";; contact object pairs:~%")
+        (dolist (c (send res :contacts))
+          (warn ";; ~A~%" (list (send c :contact_body_1) (send c :contact_body_2)))))
```

Comment:
please use `ros::ros-warn` instead of `warn`.

Reply:
Thank you, I updated.

Warn logs are like below:
```
3.irteusgl$ send *ri* :check-state-validity
[ WARN] [1605078325.760290920, 25.875000000]: Collision is found in the current state.
Contact object pairs:
(base_link bl_caster_link)
(base_link br_caster_link)
(base_link fl_caster_link)
(base_link fr_caster_link)
#<moveit_msgs::getstatevalidityresponse #X5602ef638be0>
```

Reply:
LGTM, But why caster link collides in this example?

Reply:
These collisions only occur on gazebo.
The model of the caster is different between the real fetch and gazebo.

I will create pull request for fetch.srdf to ignore these collisions in gazebo mode.
https://github.com/fetchrobotics/fetch_ros/blob/melodic-devel/fetch_moveit_config/config/fetch.srdf

real fetch urdf:
https://github.com/fetchrobotics/fetch_ros/blob/melodic-devel/fetch_description/robots/fetch.urdf
gazebo fetch urdf:
https://github.com/fetchrobotics/fetch_gazebo/blob/gazebo9/fetch_gazebo/robots/fetch.gazebo.xacro

|real|gazebo|
|---|---|
|![](https://user-images.githubusercontent.com/19769486/98897411-b278a700-24ee-11eb-94c2-a3cefd57919c.png)|![](https://user-images.githubusercontent.com/19769486/98897415-b4db0100-24ee-11eb-9283-105c170681a0.png)|

Reply:
I create https://github.com/fetchrobotics/fetch_gazebo/pull/109